crates/router/src/connector/authorizedotnet/transformers.rs

```diff
@@ -379,50 +446,141 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        let status = enums::AttemptStatus::from(item.response.transaction_response.response_code);
-        let error = item
-            .response
-            .transaction_response
-            .errors
-            .and_then(|errors| {
-                errors.into_iter().next().map(|error| types::ErrorResponse {
-                    code: error.error_code,
-                    message: error.error_text,
+        match &item.response.transaction_response {
+            Some(transaction_response) => {
+                let status = enums::AttemptStatus::from(transaction_response.response_code.clone());
+                let error = transaction_response.errors.as_ref().and_then(|errors| {
+                    errors.iter().next().map(|error| types::ErrorResponse {
+                        code: error.error_code.clone(),
+                        message: error.error_text.clone(),
+                        reason: None,
+                        status_code: item.http_code,
+                    })
+                });
+                let metadata = transaction_response
+                    .account_number
+                    .as_ref()
+                    .map(|acc_no| {
+                        Encode::<'_, PaymentDetails>::encode_to_value(
+                            &construct_refund_payment_details(acc_no.clone()),
+                        )
+                    })
+                    .transpose()
+                    .change_context(errors::ConnectorError::MissingRequiredField {
+                        field_name: "connector_metadata",
+                    })?;
+                Ok(Self {
+                    status,
+                    response: match error {
+                        Some(err) => Err(err),
+                        None => Ok(types::PaymentsResponseData::TransactionResponse {
+                            resource_id: types::ResponseId::ConnectorTransactionId(
+                                item.response
+                                    .transaction_response
+                                    .as_ref()
+                                    .map(|trans| trans.transaction_id.clone())
+                                    .unwrap_or_default(),
```

Comment:
why is it allowed to be empty? can you add comment for it

Reply:
It is should not be empty . i made necessary changes .